mlir/include/mlir/Dialect/SCF/Transforms/Transforms.h

```diff
@@ -183,8 +183,12 @@ FailureOr<ForOp> pipelineForLoop(RewriterBase &rewriter, ForOp forOp,
                                  bool *modifiedIR = nullptr);
 
 /// Create zero-trip-check around a `while` op and return the new loop op in the
-/// check. The while loop is rotated to avoid evaluating the condition twice. It
-/// turns:
+/// check. The while loop is rotated to avoid evaluating the condition twice
```

Comment:
IMO users can look into the parent op for the `if` they need.

And in `do {} while (...)` case we actually don't generate the `if` (and don't rotate the0 loop) because everything in the before block is always executed (there is a parameter to control this behavior)

I don't have a strong opinion on how this method should be designed, but only returning the loop seems to be enough for now.

Reply:
I'm going to land this as it is first. We can re-iterate on interface if needed

Reply:
Extending it as needed sounds good to me!